validation/src/main/java/io/micronaut/validation/ValidatingInterceptor.java

```diff
@@ -46,22 +57,34 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(ValidatingInterceptor.class);
 
-    private final ExecutableValidator executableValidator;
+    private final @Nullable
+    ExecutableValidator executableValidator;
+    private final ExecutableMethodValidator micronautValidator;
 
     /**
      * Creates ValidatingInterceptor from the validatorFactory.
      *
      * @param validatorFactory Factory returning initialized {@code Validator} instances
+     * @deprecated Use {@link #ValidatingInterceptor(Validator, ValidatorFactory)} instead
      */
+    @Deprecated
     public ValidatingInterceptor(Optional<ValidatorFactory> validatorFactory) {
+        this(Validator.getInstance(), validatorFactory.orElse(null));
+    }
 
-        executableValidator = validatorFactory
-                .map(factory -> factory.getValidator().forExecutables())
-                .orElse(null);
+    /**
+     * Creates ValidatingInterceptor from the validatorFactory.
+     *
+     * @param micronautValidator The micronaut validator use if no factory is available
+     * @param validatorFactory   Factory returning initialized {@code Validator} instances
+     */
+    @Inject
+    public ValidatingInterceptor(
+            Validator micronautValidator,
```

Comment:
Looks like the validator can be disabled with config. Should this be nullable?

Reply:
Yes it should